gc/base/MemoryPoolSplitAddressOrderedListBase.cpp

```diff
@@ -141,7 +141,10 @@ MM_MemoryPoolSplitAddressOrderedListBase::initialize(MM_EnvironmentBase* env)
 	 */
 	_sweepPoolManager = extensions->sweepPoolManagerSmallObjectArea;
 
-	_currentThreadFreeList = (uintptr_t*)extensions->getForge()->allocate(sizeof(uintptr_t) * _heapFreeListCount, OMR::GC::AllocationCategory::FIXED, OMR_GET_CALLSITE());
+	/* Allocate at least default _maximumHeapFreeListCount number of entries in the free list related tables. */
+	_maximumHeapFreeListCount = OMR_MAX(_maximumHeapFreeListCount, _heapFreeListCount);
+
+	_currentThreadFreeList = (uintptr_t*)extensions->getForge()->allocate(sizeof(uintptr_t) * _maximumHeapFreeListCount, OMR::GC::AllocationCategory::FIXED, OMR_GET_CALLSITE());
```

Comment:
Line exceeds 100 chars, but I've left it to be consistent with the rest of the file.